### ISSUE TITLE
[QUESTION]: Structure identifier column in data sets

### ISSUE THREAD

**Issue by Evert-Homan** [chemprop/chemprop#809]
Hi,

It seems it is not currently supported to have a compound/structure identifier column in a training set, is that correct? If not I think this would be a useful enhancement.

BW/Evert

**Comment by JacksonBurns** [comment]
Could you give an example of what you would like to see added?

**Comment by P4j45** [comment]
smiles,activity,id
c1ccccc1,5,abc

**Comment by P4j45** [comment]
Maybe it would require a flag when training or doing hyperparameter optimization, e.g. --structure_id <column_number>

Allowing for an identifier column makes it much easier to find compounds when training and predicting.

**Comment by JacksonBurns** [comment]
I understand the idea of having extra columns in the input file (this is already allowed), but is the suggestion to include these columns in the output files as well, like when using `--preds_path` to save the predictions to a file?

**Comment by davidegraff** [comment]
We probably want to flip the script on that with an `--ignore-cols` argument. That is during training and prediction, write a file that's _the same_ as the input CSV with new columns: `split` (training only) and `prediction_{0..T}`, where `T` is the number of prediction tasks. Depending on the `--ignore-cols` value, skip those columns when writing the output CSV. The assumption here is that _all_ input information in the input CSV is there for a reason and a user should have some reason to exclude it.

**Comment by JacksonBurns** [comment]
I'm more on team 'don't implement this at all' - one can map the outputs from chemprop back to the input data without us having to do it...

```python
my_input = pd.read_csv('my_data.csv', index_col='smiles')
chemprop_output = pd.read_csv('results.csv', index_col='smiles')
combined = pd.concat([my_input , chemprop_output ], axis=1)
combined.to_csv('labeled_results.csv')
```

**Comment by Evert-Homan** [comment]
chemprop_predict appears to tolerate an identifier column, which is also reported in the output. This is useful because one usually wants to do predictions on compounds not part of the training set. chemprop_train on the other hand does not tolerate an identifier column (or so it seems at least). I would like to argue that this still would be useful in order to be able to identify predictions for specific compounds in the training set.

**Comment by JacksonBurns** [comment]
Predictions on the training data generally aren't meaningful. You should use cross-validation to get predictions on every molecule in the dataset by giving each of them a chance to be in the holdout set.

If you really wanted, you could train your model and then run `chemprop_predict` on the entire dataset again, then use my code above to collate the results. But I stress, this isn't meaningful since the model was fit to this data during training.

**Comment by Evert-Homan** [comment]
I always do cross-validation when training. But it is common practice in the QSAR world to plot predicted vs experimental values for training as well as test sets. For the former you would need predictions on the training set. I'll leave it at that and work around without identifiers.

BW/Evert

**Comment by davidegraff** [comment]
+1 to @Evert-Homan . Performance on the training set is a good diagnostic to run. It's not a given that your model does well, and even in the case of vanishing train/val loss, it's valuable to investigate cases where predictions are not good on the training set.
